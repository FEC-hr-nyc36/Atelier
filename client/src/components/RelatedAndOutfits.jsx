import React from 'react';
import RelatedItems from './RelatedItems';
import Outfits from './Outfits';
import Comparing from './Comparing';

function RelatedAndOutfits () {

  return (
    <div>
      <RelatedItems />
      <Outfits />
      <Comparing />
    </div>
  );
}

export default RelatedAndOutfits;
