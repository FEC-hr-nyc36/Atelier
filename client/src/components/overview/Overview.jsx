import React, {useState, useEffect} from 'react';
import { useSelector} from 'react-redux';
import ImageGallery from './ImageGallery';
import ProductDescription from './ProductDescription';
import ProductInfo from './ProductInfo';
import StyleSelector from './StyleSelector';
import AddToCart from './AddToCart';
import ExpandedView from './ExpandedView';
import {withRouter} from 'react-router-dom';

function Overview (props) {

  const currentId = useSelector(({ currentProductId }) => currentProductId);
  const currentProduct = useSelector(({products}) => products[currentId]);
  const [currentStyle, setCurrentStyle] = useState(0);
  const [currentPhoto, setCurrentPhoto] = useState(0);

  useEffect(() => {
    props.history.push('/products/' + currentId);
  },[currentId]);
  const onStyleChange = (e) => {
    if (e.target.id !== currentStyle) {
      setCurrentPhoto(0);
      setCurrentStyle(e.target.id);
    }
  };
  const toggleExpanded = () => {
    var modal = document.getElementsByClassName('expandedModal')[0];
    var currentSlide = document.getElementsByClassName('currentSlide')[0];
    if (!modal.style.display || modal.style.display === 'none') {
      modal.style.display = 'block';
    } else {
      modal.style.display = 'none';
    }
    currentSlide.onclick = () => {
      if (!modal.style.display) {
        modal.style.display = 'block';
      }
    };
  };

  if (currentId === null) {
    return <span>loading product info</span>;
  } else {
    return (
      <div>
        <p>current id: {currentId}</p>
        {<ExpandedView
          photos={currentProduct.styleList[currentStyle].photos}
          changePhoto={setCurrentPhoto}
          currentPhoto={currentPhoto}
          changeView={toggleExpanded}/>}

        <ImageGallery
          photos={currentProduct.styleList[currentStyle].photos}
          imageClick={toggleExpanded}
          changePhoto={setCurrentPhoto}
          currentPhoto={currentPhoto}/>

        <div>Product Info Box here:

        This is where product info goes<br/>
        <ProductInfo
          product={currentProduct}
          category={currentProduct.category}
          name={currentProduct.name}
          price={currentProduct.styleList[currentStyle].original_price}
          salePrice={currentProduct.styleList[currentStyle].sale_price}/>

        This is where style selector goes<br/>
        <StyleSelector
          styleList={currentProduct.styleList}
          currentStyle={currentStyle}
          onStyleChange={onStyleChange}/>

        This is where add to cart goes
        <AddToCart
          currentStyle={currentProduct.styleList[currentStyle]}/>
        </div>

        <p>This is where product description goes</p>
        <ProductDescription
          description={currentProduct.description}
          features={currentProduct.features}/>
      </div>
    );
  }

}

export default withRouter(Overview);