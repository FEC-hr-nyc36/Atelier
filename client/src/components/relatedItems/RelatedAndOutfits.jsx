import React, { useEffect, useState } from 'react';
import { useSelector, shallowEqual } from 'react-redux';
import RelatedItems from './RelatedItems';
// import Outfits from './Outfits';
import Comparing from './Comparing';
import useRelatedProductsIds from './custom/useRelatedProductsIds';

function RelatedAndOutfits () {

  const currentProdId = useSelector((state) => state.currentProductId);
  const products = useSelector((state) => state.products, shallowEqual);
  const relatedProductsIds = useRelatedProductsIds(currentProdId, products);
  const [toggleComparing, setToggleComparing] = useState('fade-out');
  const [comparedProducts, setComparedProducts] = useState({});

  // Not memoized for access to Products
  const handleComparingToggle = (relatedId) => {
    setComparedProducts({
      current: products[currentProdId],
      related: products[relatedId]
    });
  };

  useEffect(() => {
    // Closes modal on outside of modal click
    if (toggleComparing.match(/fade-out/)) return;
    function closeModal() { setToggleComparing('fade-out'); }
    document.body.addEventListener('click', closeModal, false);
    return () => document.body.removeEventListener('click', closeModal, false);
  }, [comparedProducts]);

  return (
    <div id="related-items-and-outfits-component">
      <h2>RELATED PRODUCTS</h2>
      <RelatedItems
        relatedProductsIds={relatedProductsIds}
        products={products}
        setToggleComparing={setToggleComparing}
        handleComparingToggle={handleComparingToggle}
      />
      {/* <h2>YOUR OUTFIT</h2>
      <Outfits /> */}
      <div id="comparing-position">
        <Comparing
          visibility={toggleComparing}
          comparedProducts={comparedProducts}
        />
      </div>
    </div>
  );
}

export default RelatedAndOutfits;

