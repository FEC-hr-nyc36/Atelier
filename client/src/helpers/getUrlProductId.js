export default (pathname) => {
  var id = pathname.slice(10);
  if (id[id.length - 1] === '/') {
    id = id.slice(0, id.length - 1);
  }
  id = parseInt(id);
  return id;
};