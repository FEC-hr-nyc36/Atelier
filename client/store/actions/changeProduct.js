import axios from 'axios'
const URL = 'http://localhost:3000/'

// This is a placeholder for the makeAVERAGE function
function getAverage (nums) {
  return 3
}

function changeProduct(id) {

  return (dispatch, getState) => {
    if (getState().products[id]) {
      return dispatch({ type: 'CHANGE_PRODUCT', payload:  { id } })
    }

    Promise.all([
      axios.post(URL + 'product/', { id }),
      axios.post(URL + 'styles/', { id }),
      axios.post(URL + 'meta/', { id }),
      axios.post(URL + 'related/', { id })
  ])
  .then((data) => {
      const result = data.reduce((accum, { data }, i) => {
        if (i === 0) {
          return {
            category: data.category,
            id: data.id,
            description: data.description,
            slogan: data.slogan,
            features: data.features,
            name: data.name
          }
        } else if (i === 1) {
          return { ...accum, styleList: data.results }
        } else if (i === 2) {
          return {...accum, avgRating: getAverage(data.ratings)}
        } else if (i === 3) {
          return {...accum, relatedItemsIds: data}
        } else {
          return accum;
        }
      }, {})
      dispatch({ type: 'CHANGE_PRODUCT', payload: { [id]: result } })
    }).catch((err) => {
      console.error(err)
      dispatch({ type: 'ERROR', payload: getState().currentProductId})
    })
  }
}

export default changeProduct