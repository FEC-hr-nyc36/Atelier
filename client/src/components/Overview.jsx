import React from 'react';
import { useSelector, useDispatch} from 'react-redux'
import changeProduct from '../../store/actions/changeProduct'
import ImageGallery from './ImageGallery'
import ProductDescription from './ProductDescription'
import ProductInfo from './ProductInfo'
import StyleSelector from './StyleSelector'
import AddToCart from './AddToCart'

function Overview (props) {
  const currentId = useSelector(({ currentProductId }) => currentProductId);
  const currentProduct = useSelector(({products}) => products[currentId]);
  const [currentStyle, setCurrentStyle] = React.useState(0);
  const onStyleChange = (styleId) => {
    setCurrentStyle(styleId);
  }
  const dis = useDispatch();

  // if currentId === null

  React.useEffect(() => {
    dis(changeProduct(11003));
  },[])

  if (currentId === null) {
    return <span>loading product info</span>
  } else {
    return (
      // Image Gallery component
      // To the right of Image Gallery: div/container for :
        // Product Info Component
        // Style Selector Component
        // Add To Cart Component
      // Underneath Image Gallery and Product Info Container:
      // Product Description Component
    <div>
      <p>current id: {currentId}</p>

      <ImageGallery photos={currentProduct.styleList[currentStyle].photos}/>

      <div>Product Info Box here:

        This is where product info goes<br/>
        <ProductInfo
        rating={currentProduct.rating}
        category={currentProduct.category}
        name={currentProduct.name}
        price={currentProduct.styleList[currentStyle].original_price}
        salePrice={currentProduct.styleList[currentStyle].sale_price}/>

        This is where style selector goes<br/>
        <StyleSelector
        styleList={currentProduct.styleList}
        currentStyle={currentStyle}
        onStyleChange={onStyleChange}/>

        This is where add to cart goes
        <AddToCart
        currentStyle={currentProduct.styleList[currentStyle]}/>
      </div>

      <p>This is where product description goes</p>
      <ProductDescription
      description={currentProduct.description}
      features={currentProduct.features}/>
    </div>
    )
  }

}

export default Overview;