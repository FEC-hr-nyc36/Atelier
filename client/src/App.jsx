import React, {useEffect} from 'react';
import Overview from './components/overview/Overview';
import RelatedAndOutfits from './components/relatedItems/RelatedAndOutfits';
import { useSelector, useDispatch} from 'react-redux';
import changeProduct from '../store/actions/changeProduct';
import { BrowserRouter, Route, Switch, withRouter} from 'react-router-dom';
import getUrlProductId from './helpers/getUrlProductId';

function App(props) {
  const currentId = useSelector(({ currentProductId }) => currentProductId);
  const dis = useDispatch();
  const id = getUrlProductId(props.location.pathname);

  useEffect(() => {
    if(props.location.pathname.length > 1) {
      dis(changeProduct(id));
    } else {
      dis(changeProduct(11003));
    }
  },[]);

  if (currentId) {
    return (
      <BrowserRouter>
        <Switch>
          <Route path='/'>
            <div className="logo">
              <h1>Haworthia</h1>
            </div>

            <div className="overview-container">
              <Overview/>
            </div>
            <section className="col-center">
              <RelatedAndOutfits />
            </section>
          </Route>
        </Switch>
      </BrowserRouter>
    );
  } else {
    return null;
  }
}


export default withRouter(App);
