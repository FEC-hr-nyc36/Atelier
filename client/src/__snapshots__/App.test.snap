// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match its snapshot 1`] = `
<Fragment>
  <Overview />
  <section
    className="col-center"
  >
    <RelatedAndOutfits />
  </section>
</Fragment>
`;
